api/src/api/users/users.controllers.ts

```diff
@@ -47,4 +73,37 @@ const GetUserById = async (req: Request, res: Response, next: NextFunction) => {
   }
 };
 
-export { GetUserByEmail, GetUserById };
+const UpdateUser = async (req: RequestWithToken, res: Response, next: NextFunction) => {
+  try {
+    const payloadId = req.payload?.id || "";
+    const paramId = req.params.id;
+    if (paramId !== payloadId) throw new HttpError("user not authenticated", 500);
+    const { displayName, email } = req.body.formState;
```

Comment:
```suggestion
    const { displayName, email } = req.body;
```
No need `formState` here, just make sure below when you are sending, use `...formState` to destructure it